egs/librispeech/asr/simple_v1/train.py

```diff
@@ -99,29 +102,42 @@ def get_objf(batch: Dict,
     # nnet_output is [N, C, T]
     nnet_output = nnet_output.permute(0, 2, 1)  # now nnet_output is [N, T, C]
 
-    decoding_graph = graph_compiler.compile(texts).to(device)
```

Comment:
At this point, I believe you should be doing something like;
  P.set_scores_stochastic(model.P_scores)
Needs to be set on each iteration so the autograd graph is created...
each time the model's P_scores will be changed.

Reply:
https://github.com/k2-fsa/snowfall/blob/33cf6fedb071e31fdabbe11dbc39d9adfe8a8ada/egs/librispeech/asr/simple_v1/train.py#L194-L200

The above is where `P.scores` is normalized and set.